ihmc-perception/src/test/java/us/ihmc/perception/cuda/CUDASimpleKernelsTest.java

```diff
@@ -0,0 +1,209 @@
+//      cudaMemcpyAsync(cpuResult, gpuResult, cpuResult.sizeof(), cudaMemcpyDefault, stream);
+      cudaStreamSynchronize(stream);
+
+      int expectedValue = 5; // This is the value passed into the kernel as the first parameter in (kernel.withInt(5))
+      assertEquals(expectedValue, cpuResult.get(0), "The expected value is: " + expectedValue + " but the actual result is: " + cpuResult.get(0));
+
+      cudaFreeAsync(gpuResult, stream);
+
+      gpuResult.close();
+   }
+
+   /**
+    * Test that when you pass in the wrong type into the GPU kernel that it fails.
+    * WHen the kernel expects and int and you pass a in float, it should fail.
+    */
+   // TODO this test should fail but seems to pass even when the wrong paramter is passed into the kernel
+   @Disabled
+   @Test
+   public void testPassingInWrongTypeToGPU()
+   {
+      URL kernelPath = getClass().getResource("CUDASimpleKernels.cu");
+      stream = CUDAStreamManager.getStream();
+      program = new CUDAProgram(kernelPath);
+      kernel = program.loadKernel("pass_in_int");
+
+      kernel.withFloat(5.0f);
+
+      // TODO how do we handle the case where the kernel can run but the parameters are wrong?
+      // TODO is that something we can change or is that just the way it is with gpu programming?... (someone investigate this)
+      kernel.run(stream, new dim3(), new dim3(), 0);
+
+      //      RuntimeException thrown = assertThrows(RuntimeException.class, () -> kernel.run(stream, new dim3(), new dim3(), 0));
+      //      assertTrue(thrown.getMessage().contains("cudaErrorInvalidValue"));
+
+      cudaStreamSynchronize(stream);
+   }
+
+   /**
+    * Here the goal is to pass in the wrong number of variables and see that the GPU crashes accordingly.
+    * This test shows what happens when you make that mistake.
+    */
+   // TODO: this test fails when releasing the stream. That seems wrong because there are 6 parameters being passed into the kernel
+   // TODO: So it would seem like the kernel should not even run, but it does... (this could use some investigating)
+   // TODO: This also causes other tests to fail when this one doesn't release the steam correclty... bugs in the code lol
+   @Disabled
+   @Test
+   public void testWrongNumberOfKernelVariables()
+   {
+      URL kernelPath = getClass().getResource("CUDASimpleKernels.cu");
+      stream = CUDAStreamManager.getStream();
+      program = new CUDAProgram(kernelPath);
+      kernel = program.loadKernel("pass_in_variable");
+
+      // Pass in way to many variables and see what happens
+      kernel.withInt(5).withFloat(4.0f).withLong((long) 2.0);
+      kernel.withInt(5).withFloat(4.0f).withLong((long) 2.0);
+
+      // TODO how is this not failing when the kernel is trying to pass in a lot of variables that don't get defined in the kernel on the GPU
+      kernel.run(stream, new dim3(), new dim3(), 0);
+
+      cudaStreamSynchronize(stream);
```

Comment:
This might report the error (probably check for the runtime exception here?)
```suggestion
      CUDATools.checkCUDAError(cudaStreamSynchronize(stream));
```

Reply:
Ok I'll check that out